library/common/extensions/filters/http/network_configuration/config.cc

```diff
@@ -0,0 +1,30 @@
+#include "library/common/extensions/filters/http/network_configuration/config.h"
+
+#include "library/common/extensions/filters/http/network_configuration/filter.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace HttpFilters {
+namespace NetworkConfiguration {
+
+Http::FilterFactoryCb NetworkConfigurationFilterFactory::createFilterFactoryFromProtoTyped(
+    const envoymobile::extensions::filters::http::network_configuration::NetworkConfiguration&,
+    const std::string&, Server::Configuration::FactoryContext& context) {
+
+  auto network_configurator = Network::ConfiguratorHandle{context}.get();
```

Comment:
This pattern looks a little bit weird (at first thought I thought ConfigurationHandle was a RAII object, and that we threw it away immediately after getting the inner object). I probably would have opted for a single static function to construct/return the singleton instead of a temporary object

Non-blocking

Reply:
I lifted this pattern from some examples in Envoy, though in Envoy these tended to be called `Factory` rather than `Handle`. Would that be clearer?

Reply:
Yeah I think that would make it clearer, thanks!